game/src/contrib/entities/AIFactory.java

```diff
@@ -0,0 +1,215 @@
+package contrib.entities;
+
+import contrib.utils.components.ai.fight.CollideAI;
+import contrib.utils.components.ai.fight.RangeAI;
+import contrib.utils.components.ai.idle.PatrouilleWalk;
+import contrib.utils.components.ai.idle.RadiusWalk;
+import contrib.utils.components.ai.idle.StaticRadiusWalk;
+import contrib.utils.components.ai.transition.ProtectOnApproach;
+import contrib.utils.components.ai.transition.ProtectOnAttack;
+import contrib.utils.components.ai.transition.RangeTransition;
+import contrib.utils.components.ai.transition.SelfDefendTransition;
+import contrib.utils.components.skill.FireballSkill;
+import contrib.utils.components.skill.Skill;
+import contrib.utils.components.skill.SkillTools;
+import core.Entity;
+import core.Game;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Random;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+
+/**
+ * A Class solely dedicated to randomly generating all types of AIs.
+ *
+ * NOTE: Random ints get a +1 inside the random function due to the exclusionary nature of random.nextInt()
+ * NOTE: If the first Monster rolls a ProtectOn AI, it will Protect itself.
+ */
+public class AIFactory {
+    //FightAI Parameters:
+    //CollideAI
+    private static final float rushRangeLow = 0.5f;
```

Comment:
Konstanten werden in Java immer UPPERCASE mit "_" als Seperator geschrieben.
Also `RUSH_RANGE_LOW`

Reply:
Hab ich wohl nicht dran gedacht, ist gefixt.